pkg/controller/usersignupcleanup/usersignup_cleanup_controller_test.go

```diff
@@ -143,6 +146,29 @@ func TestUserCleanup(t *testing.T) {
 		require.False(t, res.Requeue)
 	})
 
+	t.Run("test propagation policy", func(t *testing.T) {
+		userSignup := test2.NewUserSignup(
+			test2.CreatedBefore(threeYears),
+			test2.VerificationRequired(),
+		)
+
+		r, req, fakeClient := prepareReconcile(t, userSignup.Name, userSignup)
+		deleted := false
```

Comment:
this is not needed, I believe. What really matters in this test (as you did) is verifying that the object is deleted with the `DeletePropagationForeground`